src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -149,8 +149,11 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
+			IntegrationFlow theFlow = this.registry.remove(flowId);
```

Comment:
Thank you for catching the typo!
I'm not sure about your simplicity suggestion. 
End-user can choose `autoStartup = false` mode. So, we won't see this `IllegalStateException`.
In that case it is his (her) own responsibility to search for all endpoints and start them manually. 
That's why I'm pointing there an `autoStartup` for distinguishing the difference in behavior.